core/domain/feedback_services.py

```diff
@@ -234,46 +237,63 @@ def get_suggestion(exploration_id, thread_id):
             exploration_id, thread_id))
 
 
-def get_open_threads(exploration_id, has_suggestion):
+def _get_thread_from_model(thread_model):
+    return feedback_domain.FeedbackThread(
+        thread_model.id, thread_model.exploration_id, thread_model.state_name,
+        thread_model.original_author_id, thread_model.status,
+        thread_model.subject, thread_model.summary, thread_model.has_suggestion,
+        thread_model.created_on, thread_model.last_updated)
+
+
+def get_threads(exploration_id):
+    thread_models = feedback_models.FeedbackThreadModel.get_threads(
+        exploration_id)
+    threads = []
+    for model in thread_models:
+        threads.append(_get_thread_from_model(model))
+    return threads
+
+
+def get_displayable_open_threads(exploration_id, has_suggestion):
     """If has_suggestion is True, return a list of all open threads that have a
     suggestion, otherwise return a list of all open threads that do not have a
     suggestion."""
 
-    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    threads = get_threads(exploration_id)
     open_threads = []
     for thread in threads:
         if (thread.has_suggestion == has_suggestion and
                 thread.status == feedback_models.STATUS_CHOICES_OPEN):
             open_threads.append(thread)
     return [
-        _get_thread_dict_from_model_instance(t)
+        _get_thread_dict(t)
         for t in open_threads]
 
 
-def get_closed_threads(exploration_id, has_suggestion):
+def get_displayable_closed_threads(exploration_id, has_suggestion):
     """If has_suggestion is True, return a list of all closed threads that have
     a suggestion, otherwise return a list of all closed threads that do not have
     a suggestion."""
 
-    threads = feedback_models.FeedbackThreadModel.get_threads(exploration_id)
+    threads = get_threads(exploration_id)
     closed_threads = []
     for thread in threads:
         if (thread.has_suggestion == has_suggestion and
                 thread.status != feedback_models.STATUS_CHOICES_OPEN):
             closed_threads.append(thread)
     return [
-        _get_thread_dict_from_model_instance(t)
+        _get_thread_dict(t)
         for t in closed_threads]
 
 
-def get_all_threads(exploration_id, has_suggestion):
+def get_displayable_threads(exploration_id, has_suggestion):
     """Return a list of all threads with suggestions."""
```

Comment:
Hm, looks like the (old) docstring here is wrong -- it should be "Returns a list of all threads with suggestions if has_suggestion is True; otherwise, returns a list of all threads with no suggestions."